plugins/pencilblue/include/routes.js

```diff
@@ -68,22 +68,36 @@ module.exports = function Routes(pb){
             controller: path.join(pb.config.docRoot, 'plugins', 'pencilblue', 'controllers', 'admin', 'login.js'),
             content_type: 'text/html'
         },
+        {
+            method: 'post',
+            path: "/salesforce/change_password",
+            auth_required: true,
+            inactive_site_access: true,
+            controller: path.join(pb.config.docRoot, 'plugins', 'pencilblue', 'controllers', 'actions', 'salesforce', 'change_password.js'),
+            content_type: 'application/json'
+        },
+        {
+            method: 'patch',
+            path: "/salesforce/change_email",
+            auth_required: true,
+            inactive_site_access: true,
+            controller: path.join(pb.config.docRoot, 'plugins', 'pencilblue', 'controllers', 'actions', 'salesforce', 'change_email.js'),
+            content_type: 'application/json'
+        },
         {
             method: 'get',
             path: "/login/salesforce",
-            access_level: 0,
             auth_required: false,
```

Comment:
Because auth is not required, I do not recall what context would be found in controller above (i.e. when you call `this.ctx.session.authentication.user` in the code above).  Perhaps @lgenet can chime in better here.  And as a best practice point of view, I think we should put the routes in controllers themselves rather than a big routes file.

Reply:
Routes in Controllers:
- It is the convention of the Pencilblue plugin to have all routes in the routes file instead of in the controller. 
- I dont know if you can actually put them in the file - I dont know if the getRotues hook fires over controllers in Pencilblue plugin since it is not initialized like the others.  sooo until we get a replacement admin plugin, we are stuck doing this.  

Access Level and Context:
- I believe, you should have the user context from the cookie.  Your cookie has a session ID on it that we use to look up session in redis and stamp onto context during the middleware stack.  If your auth level does not match that which is required for the route we throw a 401 error but otherwise we let it through.  If one is not found, we give you a blank/default auth object.